src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -978,7 +991,17 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                             }
 
                             utf16Data = Vector128.CreateScalarUnsafe(possibleNonAsciiQWord).AsInt16();
-                            Unsafe.WriteUnaligned<uint>(pOutputBuffer, Sse2.ConvertToUInt32(Sse2.PackUnsignedSaturate(utf16Data, utf16Data).AsUInt32()));
+
+                            if (AdvSimd.IsSupported)
+                            {
+                                Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
+                                Vector128<byte> source = AdvSimd.ExtractNarrowingSaturateUnsignedUpper(lower, AdvSimd.LoadVector128((short*)pInputBuffer + Vector128<short>.Count));
```

Comment:
>source [](start = 48, length = 6)

I am not too familiar with `ExtractNarrowingSaturateUnsigned*` APIs and @Egor.Chesakov@microsoft.com  should be the one to approve it, but I am not sure if `source` is accurate name for this operation.

Reply:
I see what you mean. Maybe I should name it result, that would match the intrinsic method: https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vqmovun_high_s16

Reply:
Also, the second argument of *Upper should be utf16data, from what I've seen in other usage examples.

Reply:
You don't need call to ExtractNarrowingSaturateUnsignedUpper here at all for the same reason as in https://github.com/dotnet/runtime/pull/39041/files#r454715723. 

The following should be equivalent to Sse2 code:
```c#
Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
AdvSimd.StoreSelectedScalar((uint*)pOutputBuffer, lower.AsUInt32(), 0);
```